upgrade-2.x.sh

```diff
@@ -249,6 +250,176 @@ function device_type_match {
     echo "${match}"
 }
 
+# Pre update cleanup: remove some not-required files from the boot partition to clear some space
+function pre_update_pi_bootfiles_removal {
+    local boot_files_for_removal=('start_db.elf' 'fixup_db.dat')
+    for f in "${boot_files_for_removal[@]}"; do
+        echo "Removing $f from boot partition"
+        rm -f "/mnt/boot/$f"
+    done
+    sync
+}
+
+function pre_update_fix_bootfiles_hook {
+    log "Applying bootfiles hostapp-hook fix"
+    local bootfiles_temp
+    bootfiles_temp=$(mktemp)
+    curl -f -s -L -o "$bootfiles_temp" https://raw.githubusercontent.com/resin-os/resinhup/master/upgrade-patches/0-bootfiles || log ERROR "Couldn't download fixed '0-bootfiles', aborting."
+    chmod 755 "$bootfiles_temp"
+    mount --bind "$bootfiles_temp"  /etc/hostapp-update-hooks.d/0-bootfiles
+}
+
+function hostapp_based_update {
+    local update_package=$1
+    local storage_driver
+    local sysroot="/mnt/sysroot/inactive"
+
+    case ${SLUG} in
+        raspberry*)
+            log "Running pre-update fixes for ${SLUG}"
+            pre_update_pi_bootfiles_removal
+            if ! version_gt "${VERSION_ID}" "2.7.6" ; then
+                pre_update_fix_bootfiles_hook
+            fi
+            ;;
+        *)
+            log "No device-specific pre-update fix for ${SLUG}"
+    esac
+
+    if ! [ -S "/var/run/docker-host.sock" ]; then
+        ## Happens on devices booting after a regular HUP update onto a hostapps enabled resinOS
+        log "Do not have docker-host running"
+        log "Clean inactive partition"
+        rm -rf "${sysroot:?}/"*
+        log "Getting storage driver info"
+        storage_driver=$(cat /boot/storage-driver)
+        log "Starting docker-host with ${storage_driver} storage driver"
+        dockerd --log-driver=journald -s="${storage_driver}" --data-root="${sysroot}/docker" -H unix:///var/run/docker-host.sock --pidfile=/var/run/docker-host.pid --exec-root=/var/run/docker-host --bip 10.114.101.1/24 --fixed-cidr=10.114.101.128/25 --iptables=false &
+        sleep 10
+    else
+        if [ -f "$sysroot/resinos.fingerprint" ]; then
+            # Happens on a device, which has HUP'd from a non-hostapp resinOS first
+            # then updated again but this time with hostapp-update. The previous "active"
+            # partition now inactive, and still has leftover data
+            log "Have docker-host running, with dirty inactive partition"
```

Comment:
How can this happen? In a hostapp to hostapp update there should be no garbage left behind that need to be cleaned up after the new OS has booted

Reply:
@petrosagg it's not the hostapp-to-hostapp updatre case, but as the header says, that non-hostapp-to-hostapp. There the previous system is still there, as before the reboot it was running, and currently don't have a service in resinOS that clears up `/mnt/sysroot/inactive` on boot (before starting the `docker-host` service).

Reply:
Right. I thought I had deleted this comment before submitting my review. Thanks for clarifying anyway